### PR TITLE
Authentication Tutorial

### PR DESCRIPTION
**Pull request by InfernalHydra**
Pretty long and lots of code.  Everything should work, let me know if anything does not work and I will try to see if I can replicate the issue and fix it.

### CODE REVIEW

**Comment by kdesai2018** [comment]
from what i tell this looks phenomenal :) great work ryan

File: authentication/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Authentication 
+## Introduction
+
+This tutorial will introduce the basics of authentication, implemented for both websites and mobile applications.  The code and ideas presented here **should only be used in a hackathon environment**.  In a production environment, most of the tatics listed here would not be secure enough, and may leave ways for determined hackers to access sensitive information.  The code here utilizes mostly Javascript and MongoDB.
```
Comment (didizhou1) [inline authentication/README.md:4]: Can you add a quick sentence on what authentication is and when you should worry about adding it to a project and when it's not necessary? (though should be pretty obvious)

Comment (InfernalHydra) [reply authentication/README.md:4]: Oops, I didn't see that I didn't finish the paragraph.  I will add everything requested soon.

**Comment by InfernalHydra** [comment]
I just added all of the changes requested by Didi.  Please go through the documents and make sure that everything sounds good and checks out!

**Pull request merged**